data/src/main/scala/io/prediction/data/api/EventServer.scala

```diff
@@ -96,18 +100,34 @@ class EventServiceActor(
             }.getOrElse{
               Right(AuthData(k.appid, None))
             }
-          }.getOrElse{
-            Left(AuthenticationFailedRejection(
-              AuthenticationFailedRejection.CredentialsRejected, List()))
-          }
-        }.getOrElse { Left(AuthenticationFailedRejection(
-          AuthenticationFailedRejection.CredentialsMissing, List()))
+          }.getOrElse(FailedAuth)
+        }.getOrElse {
+          // with accessKey in header, return appId if succeed
+          ctx.request.headers.find(_.name == "Authorization").map { authHeader ⇒
+            authHeader.value.split("Basic ") match {
+              case Array(_, value) ⇒
+                val appAccessKey =
+                  new String(base64Decoder.decodeBuffer(value)).trim.split(":")(0)
+                accessKeysClient.get(appAccessKey) match {
+                  case Some(k) ⇒ Right(AuthData(k.appid, None))
+                  case None ⇒ FailedAuth
+                }
+
+              case _ ⇒ FailedAuth
+            }
+          }.getOrElse(FailedAuth)
```

Comment:
there is "CredentialsMissing" case in addition to "CredentialsRejected" case in original code. would you add a test in for this in the SegmenIOAuthSpec too?